### DISCUSSION TITLE
Certificate error using OAuth2 on SvelteKit (tls: failed to verify certificate: x509: certificate signed by unknown authority)

### DISCUSSION THREAD

**Discussion by madariagamikel** [pocketbase/pocketbase#3305] [Q&A] [answered]
Hello, 

I was wondering if someone could help me out with an issue. I've followed [this tutorial](https://www.youtube.com/watch?v=Ig_RMc0GFw4) that helps set up a web app with **Google Auth, pocketbase and sveltekit**. After downloading the [source code](https://github.com/consultingninja/pbOAuth2) and launching the app, the auth doesn't work properly. It redirects me correctly but it doesn't register the user in the pocketbase user collection. 

This is the error I'm getting in the logs:

`
{   "errorMessage": "Post \"https://accounts.google.com/o/oauth2/token\": tls: failed to verify certificate: x509: certificate signed by unknown authority" }`

![1](https://github.com/pocketbase/pocketbase/assets/62753476/7ffa7a9b-058c-4835-8706-2847fad9a7c2)

I'm running the app locally without SSL and pocketbase is hosted in Docker [with this image](https://hub.docker.com/r/spectado/pocketbase).

Thanks in advance!!

**Comment by ganigeorgiev** [comment]
It seems that the above Docker image is missing the `ca-certificates` package which the base image (alpine) doesn't include by default.

You can create your own Dockerfile based on https://pocketbase.io/docs/going-to-production/#using-docker (_if you need Docker in the first place_).

**Reply by madariagamikel** [reply]
Thanks for the fast response! I just noticed that I shared the wrong docker image. [This](https://hub.docker.com/r/spectado/pocketbase) is the one I'm actually using (I've updated the main post). I've checked that it does install the `ca-certificates` library, so it seems that we are facing another issue.

**Reply by ganigeorgiev** [reply]
I can't find the source for the above Docker image. How did you check that it installs `ca-certificates`? If it is a multi-stage build, double check that `ca-certificates` is available in the final/resulting image.

_But in general, I'd advice against blindly pulling unofficial/random images especially if you are not sure what you are doing. Note that PocketBase doesn't need Docker and can be used directly without it._

**Reply by madariagamikel** [reply]
Hello! You can check that it installs `ca-certificates` you have to access [tags ](https://hub.docker.com/r/spectado/pocketbase/tags), click the [latest](https://hub.docker.com/layers/spectado/pocketbase/latest/images/sha256-04a71e383706e5837d4061403019b9d7536010b3344a399a64e47a06101f7592?context=explore) and then click the third layer (`RUN...`). There you can see the following: `RUN /bin/sh -c apk update && apk add ca-certificates && rm -rf /var/cache/apk/* # buildkit`.

I've tried running the app without docker and it works perfectly, no certificate errors (installing pocketbase in Windows). 

Then, I've created the dockerfile based on the [example](https://pocketbase.io/docs/going-to-production/#using-docker) you shared. This time the same error appears, so it seems that the problem only happens in Linux/Docker.

**Reply by ganigeorgiev** [reply]
I'm using Linux and I can't reproduce it (both with the executable and using the Dockerfile from the docs).

I guess it could be something specific to the Docker environment on Windows as I'm not really sure how that works there but I don't have a Windows VM at the moment to test it.

**Comment by madariagamikel** [comment, marked as answer]
I solved the problem (running the app via Docker) installing my organization's Root CA Certificate in the Docker image. For extracting the certificate I followed [this tutorial](https://medium.com/@sanghviyash6/how-to-get-ca-certificate-of-any-server-using-google-chrome-e8db3e4d3fcf). To install it in Docker I added the following code (see ++) to the [Dockerfile ](https://pocketbase.io/docs/going-to-production/#using-docker) provided in the PocketBase documentation: 

```
FROM alpine:latest

ARG PB_VERSION=0.18.5

RUN apk add --no-cache \
    unzip \
    ca-certificates

++ ADD ORGANIZATION_CERTIFICATE.crt /usr/local/share/ca-certificates/foo.crt
++ RUN chmod 644 /usr/local/share/ca-certificates/foo.crt && update-ca-certificates

ADD [https://github.com/pocketbase/pocketbase/releases/download/v${PB_VERSION}/pocketbase_${PB_VERSION}_linux_amd64.zip](https://github.com/pocketbase/pocketbase/releases/download/v$%7BPB_VERSION%7D/pocketbase_$%7BPB_VERSION%7D_linux_amd64.zip) /tmp/pb.zip

RUN unzip /tmp/pb.zip -d /pb/

EXPOSE 8080

CMD ["/pb/pocketbase", "serve", "--http=0.0.0.0:8080"]
```
Thanks for your help and I hope this information can help others!

**Reply by ViktorLi1988** [reply]
I am trying to follow your solution but don't understand which certificate you are using? Is it Google's GTS Root R1? Or some other certificate? Thank you.

**Reply by madariagamikel** [reply]
Hi Viktor. The name of the certificate is different depending on your organization, but in any case you should be using the higher level (root) one.

**Reply by ViktorLi1988** [reply]
Thank you for fast reply Mikel. In the end I did use /usr/share/ca-certificates/mozilla/GTS_Root_R1.crt file from the machine that hosts PB docker container. I copied the file in the same directory as PB containers Dockerfile and applied the code you suggested inside Dockerfile like this:
```
++ ADD GTS_Root_R1.crt /usr/local/share/ca-certificates/foo.crt
++ RUN chmod 644 /usr/local/share/ca-certificates/foo.crt && update-ca-certificates
```
Everything works fine now.
Thank you once again!
